pkg/controller.v1/pytorch/pytorchjob_controller.go

```diff
@@ -525,3 +536,39 @@ func (r *PyTorchJobReconciler) onOwnerCreateFunc() func(createEvent event.TypedC
 		return true
 	}
 }
+
+func desiredPyTorchJobNetworkPolicy(job *kubeflowv1.PyTorchJob) *networkingv1ac.NetworkPolicyApplyConfiguration {
+	return networkingv1ac.NetworkPolicy(job.Name, job.Namespace).
+		WithLabels(map[string]string{"job-name": job.Name}).
+		WithSpec(networkingv1ac.NetworkPolicySpec().
+			WithPodSelector(metav1ac.LabelSelector().WithMatchLabels(map[string]string{"job-name": job.Name})).
```

Comment:
Default label created by KFTO for PyTorchJob is `training.kubeflow.org/job-name`.
This pod selector doesn't match it, so PyTorchJobs without explicitly specified label `job-name` are not affected by this NetworkPolicy.